users/test.py

```diff
@@ -36,36 +40,139 @@ def setUp(self):
         """An authorized user should receive authorized privileges."""
-        self.assertTrue(find_authorization_in_db(
+        self.assertTrue(app.find_authorization_in_db(
             AUTHORIZED_USER, self.mock_collection))
 
     def test_non_authorized_user_not_authorized(self):
         """A non-authorized user should not receive authorized privileges."""
-        self.assertFalse(find_authorization_in_db(
+        self.assertFalse(app.find_authorization_in_db(
             NON_AUTHORIZED_USER, self.mock_collection))
 
     def test_malformatted_user_none_authorization(self):
         """An incorrect db schema should not receive authorized privileges."""
-        self.assertFalse(find_authorization_in_db(
+        self.assertFalse(app.find_authorization_in_db(
             MISSING_USER, self.mock_collection))
 
     def test_unkown_user_not_authorized(self):
         """A user not in the db should not receive authorized privileges."""
-        self.assertFalse(find_authorization_in_db(
+        self.assertFalse(app.find_authorization_in_db(
             MALFORMATTED_IN_DB_USER, self.mock_collection))
 
 
+class TestUserUpsertion(unittest.TestCase):
+    """Test updating/inserting users into the db."""
+
+    def test_insert_valid_user(self):
+        """A valid user object should be upserted and retrieved correctly."""
+        self.mock_collection = mongomock.MongoClient().db.collection
+        user_to_insert = {
+            "user_id": USER_ID,
+            "name": USER_NAME
+        }
+        app.upsert_user_in_db(user_to_insert, self.mock_collection)
+        found_user = self.mock_collection.find_one({"user_id": USER_ID})
+        self.assertEqual(user_to_insert["user_id"], found_user["user_id"])
+        self.assertEqual(user_to_insert["name"], found_user["name"])
+
+    def test_user_auth_defaults_are_set(self):
+        """app.upsert_user_in_db should set any default value before upsertion.
+
+        Right now, only the 'is_organizer' field of an inserted user should
+        default to False.
+        """
+        self.mock_collection = mongomock.MongoClient().db.collection
+        user_to_insert = {
+            "user_id": USER_ID,
+            "name": USER_NAME
+        }
+        app.upsert_user_in_db(user_to_insert, self.mock_collection)
+        found_user = self.mock_collection.find_one({"user_id": USER_ID})
+        self.assertFalse(found_user["is_organizer"])
+
+    def test_malformatted_user_not_inserted(self):
+        """A malformatted user_object should not be inserted.
+
+        Note: app.upsert_user_in_db should not insert the user if it is missing
+        the 'user_id' or 'name' attributes, but should insert the user if
+        the user has more attributes than those. If the user is malformatted,
+        app.upsert_user_in_db raises an AttributeError, else it upserts the
+        user and returns the new user object.
+        """
+        self.mock_collection = mongomock.MongoClient().db.collection
```

Comment:
You might want to initialize this in `setUp`?

Reply:
Done https://github.com/knative-portability/large-events/pull/8/commits/4a27bbae23fd046b48d07e2ebd5e75b342ca2c70